activerecord/test/cases/migration/create_join_table_test.rb

```diff
@@ -78,6 +78,17 @@ def test_create_join_table_with_index
         assert_equal [%w(artist_id music_id)], connection.indexes(:artists_musics).map(&:columns)
       end
 
+      def test_create_join_table_respects_reference_key_type
+        connection.create_join_table :artists, :musics do |t|
+          t.references :video
```

Comment:
`references` key type is the same as `primary_key` type and it depends on an adapter. `bigint` for mysql2 and postgresql, `integer` for sqlite3 (related #27334). If `primary_key` is `bigint`, `create_join_table` should create `bigint` columns.